src/s2wasm.h

```diff
@@ -1193,6 +1198,42 @@ class S2WasmBuilder {
         }
       }
     }
+    if (!!startFunction) {
```

Comment:
Because it refers to a string `"\0"` which isn't `nullptr`, whereas `operator!` checks for `nullptr` **and** `str[0] == '\0'`. It's not very idiomatic C++!